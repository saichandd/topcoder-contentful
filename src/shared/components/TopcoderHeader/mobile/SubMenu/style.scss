@import '~styles/mixins';

.item {
  color: $tc-white;
  font: 500 14px "Sofia Pro", sans-serif;
  margin: 9px 18px;
  text-transform: uppercase;

  a:hover {
    color: $tc-dark-blue;
  }
}

.sub-menu {
  padding: 6px 0;
}

.title {
  color: $tc-gray-40;
  font: 500 18px "Sofia Pro", sans-serif;
  margin: 9px 18px;
  text-transform: uppercase;
}
