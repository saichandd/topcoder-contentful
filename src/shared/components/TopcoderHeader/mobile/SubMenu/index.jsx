import PT from 'prop-types';
import React from 'react';
import './style.scss';

export default function SubMenu({
  hideTitle,
  subMenu,
}) {
  const items = subMenu.items.map(item => (
    <li key={item.title} styleName="item">
      <a href={item.link}>
        {item.title}
      </a>
    </li>
  ));
  const title = hideTitle ? '' : (
    <li styleName="title">
      {subMenu.title}
    </li>
  );
  return (
    <ul styleName="sub-menu">
      {title}
      {items}
    </ul>
  );
}

export const SUB_MENU_SHAPE = PT.shape({
  title: PT.string.isRequired,
  items: PT.arrayOf(PT.shape({
    icon: PT.node.isRequired,
    link: PT.string.isRequired,
    title: PT.string.isRequired,
  })).isRequired,
});

SubMenu.defaultProps = {
  hideTitle: false,
};

SubMenu.propTypes = {
  hideTitle: PT.bool,
  subMenu: SUB_MENU_SHAPE.isRequired,
};
