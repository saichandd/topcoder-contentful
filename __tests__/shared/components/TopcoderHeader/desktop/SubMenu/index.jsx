import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import SubMenu from 'components/TopcoderHeader/desktop/SubMenu';

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <SubMenu
      closeMenu={_.noop}
      menu={{ items: [{ title: 'title', icon: 'icon url', link: 'menu url' }] }}
    />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <SubMenu {...this.props} />;
  }
}

const closeMenu = jest.fn();
const instance = TU.renderIntoDocument((<Wrapper
  trigger={{
    bottom: 10,
    left: 0,
    right: 10,
    top: 0,
  }}
  closeMenu={closeMenu}
/>));

test('closeMenu cancelled', () => {
  const menus = TU.scryRenderedDOMComponentsWithTag(instance, 'ul');
  expect(menus).toHaveLength(1);
  TU.Simulate.mouseLeave(menus[0], { pageY: 1 });
  expect(closeMenu).not.toHaveBeenCalled();
});

test('closeMenu executed', () => {
  const menus = TU.scryRenderedDOMComponentsWithTag(instance, 'ul');
  expect(menus).toHaveLength(1);
  TU.Simulate.mouseLeave(menus[0], { pageY: -1 });
  expect(closeMenu).toHaveBeenCalled();
});
