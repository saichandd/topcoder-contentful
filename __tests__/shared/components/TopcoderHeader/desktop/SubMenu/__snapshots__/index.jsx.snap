// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<ul
  className="src-shared-components-TopcoderHeader-desktop-SubMenu-___style__opened-menu___2t7di"
  onMouseLeave={[Function]}
>
  <Item
    closeMenu={[Function]}
    currentSubMenuTitle=""
    enforceA={false}
    icon="icon url"
    link="menu url"
    title="title"
  />
</ul>
`;
